File: frontend_pasien/lib/models/pasien.dart
class Pasien {
  final int pasienId;
  final String nama;
  final String email;
  final String noTelepon;
  final String jenisKelamin;
  final String tanggalLahir;
  final String alamat;

  Pasien({
    required this.pasienId,
    required this.nama,
    required this.email,
    required this.noTelepon,
    required this.jenisKelamin,
    required this.tanggalLahir,
    required this.alamat,
  });

  factory Pasien.fromJson(Map<String, dynamic> json) {
    return Pasien(
      pasienId: json['pasien_id'] ?? 0,
      nama: json['nama'] ?? '',
      email: json['email'] ?? '',
      noTelepon: json['no_telepon'] ?? '',
      jenisKelamin: json['jenis_kelamin'] ?? '',
      tanggalLahir: json['tanggal_lahir'] ?? '',
      alamat: json['alamat'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'pasien_id': pasienId,
      'nama': nama,
      'email': email,
      'no_telepon': noTelepon,
      'jenis_kelamin': jenisKelamin,
      'tanggal_lahir': tanggalLahir,
      'alamat': alamat,
    };
  }
}

File: main-backend/internal/domain/pasien.go
package domain

import "time"

type Pasien struct {
	PasienID        int        `gorm:"primaryKey;column:pasien_id"`
	Nama            string     `gorm:"column:nama;not null"`
	Email           string     `gorm:"column:email;uniqueIndex"`
	Password        string     `gorm:"column:password;not null"`
	NIK             string     `gorm:"column:nik;uniqueIndex"`
	TanggalLahir    time.Time  `gorm:"column:tanggal_lahir"`
	TempatLahir     string     `gorm:"column:tempat_lahir"`
	Alamat          string     `gorm:"column:alamat;type:text"`
	JenisKelamin    string     `gorm:"column:jenis_kelamin"`
	NoTelepon       string     `gorm:"column:no_telepon"`
	ResetCode       *string    `gorm:"column:reset_code"`
	ResetCodeExpiry *time.Time `gorm:"column:reset_code_expiry"`
	CreatedAt       time.Time  `gorm:"autoCreateTime:milli;column:created_at"`
	UpdatedAt       time.Time  `gorm:"autoUpdateTime:milli;column:updated_at"`
}

func (Pasien) TableName() string {
	return "pasien"
}
